kernel/src/main/java/org/kframework/parser/concrete2kore/ParseInModule.java

```diff
@@ -92,19 +100,51 @@ public Module seedModule() {
      * @return Module with extra productions defined during parser generator.
      */
     public Module getExtensionModule() {
-        return extensionModule;
+        Module extM = extensionModule;
+        if (extM == null) {
+            Tuple3<Module, Module, Module> mods = RuleGrammarGenerator.getCombinedGrammarImpl(seedModule);
+            extM = mods._1();
+            disambModule = mods._2();
+            parsingModule = mods._3();
+            extensionModule = extM;
```

Comment:
Can we factor out a `private void loadModules()` here which all of `getExtensionModule()`, `getParsingModule()`, and `getDisambiguationModule()` can use?

Reply:
Or perhaps `initializeModules()` to mirror the `initialize()` below.

Reply:
I mean, the implementation of the three methods is different... I'm not really sure how I would go about doing that. I'm not opposed, but I'm just unclear how that would be done.

Reply:
Unfortunately that is not possible either because in order to implement the racy single-check pattern, we need slightly different assignment statements in each of the three functions.